### PR TITLE
Fix having empty string for `rootpath` in coco format

### PR DESCRIPTION
**Pull request by cih9088**
### Summary
Datumaro coco format supports initializing from a specific annotation file.

If the given annotation file path is a relative path **not specifying current working directory**

If 1) the current working directory **IS** dataset directory and the given annotation file path is a relative path **not specifying current directory**, 2) the current working directory is **inside of dataset directory**, importing dataset would be failed.

```console
$ ls
annotations/ images/
```
```python
import os
from datumaro import Dataset

Dataset.import_from('annotations/instances_train2017.json', 'coco')    # error -> fixed in this PR
Dataset.import_from('./annotations/instances_train2017.json', 'coco')  # ok

os.chdir('annotations')
Dataset.import_from('instances_train2017.json', 'coco')                # error -> fixed in this PR
```

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added the description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md).​
- [ ] I have updated the [documentation](https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly

### License

- [ ] I submit _my code changes_ under the same [MIT License](https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Review by vinnamkim — DISMISSED** [review]
Would you add a test case for both?
```python
Dataset.import_from('annotations/instances_train2017.json', 'coco')    # error -> fixed in this PR
Dataset.import_from('./annotations/instances_train2017.json', 'coco')  # ok
```

**Comment by cih9088** [comment]
> Would you add a test case for both?

Added in https://github.com/openvinotoolkit/datumaro/pull/979/commits/6afcc9f45ce7d3c33827ea27e716ef534bd56b38

File: tests/unit/test_coco_format.py [hunk 1]
```diff
@@ -81,6 +82,50 @@ def test_can_export_and_import_back(self):
                 back_dataset = Dataset.import_from(path_test_dir)
             compare_datasets(self, dataset, back_dataset)
 
+    @mark_requirement(Requirements.DATUM_GENERAL_REQ)
+    def test_can_import_from_any_cwd(self):
```
Comment (vinnamkim) [inline tests/unit/test_coco_format.py:86]: Does this test include the "1) not specifying current directory" case?

Comment (cih9088) [reply tests/unit/test_coco_format.py:86]: Yes, `os.path.relpath` does return path without `.`.

```python
>>> import os
>>> cwd = os.getcwd()
>>> os.listdir()
['.git', '.github', '.gitmodules', 'explorer.png', 'LICENSE', 'NOTICE', 'datum.py', 'notebooks', 'pytest.ini', 'security.md', 'tests', '.markdownlint.yaml', 'ipas_default.config', 'build', 'test_progressive.py', 'pts_in_hull.npy', 'MANIFEST.in', 'README.md', 'CHANGELOG.md', 'requirements-default.txt', 'datumaro.egg-info', '.tox', 'media.py', '3rd-party.txt', '.pre-commit-config.yaml', 'contributing.md', '.pytest_cache', '.gitattributes', 'setup.py', 'requirements-dev.txt', 'colorization_deploy_v2.prototxt', 'docs', 'pyproject.toml', '.gitignore', 'datumaro', '.ruff_cache', 'cache', 'lightning_logs', 'logs', 'colorization_release_v2.caffemodel', '__pycache__', 'test.py', 'requirements.txt', 'tox.ini', 'requirements-core.txt']

>>> os.path.relpath(os.path.join(cwd, 'datumaro'), cwd)
'datumaro'
```

The first relative path import test would handle 1) case

**Review by vinnamkim — APPROVED** [review]
LGTM.

**Review by JihwanEom — APPROVED** [review]

**Review by wonjuleee — APPROVED** [review]

**Pull request merged**